tests/Unit/Evolution/Systems/GeneralizedHarmonic/Test_Constraints.cpp

```diff
@@ -894,8 +895,8 @@ void test_constraint_compute_items(
                                                              Frame::Inertial>>>(
       x, spatial_metric, lapse, shift, deriv_spatial_metric, deriv_lapse,
       deriv_shift, time_deriv_spatial_metric, time_deriv_lapse,
-      time_deriv_shift, deriv_gauge_source, time_deriv_gauge_source, deriv_phi,
-      deriv_pi);
+      time_deriv_shift, deriv_gauge_source, time_deriv_gauge_source,
+      deriv_spacetime_metric, deriv_phi, deriv_pi);
```

Comment:
(just choosing a line at random) does this resolve the issue that the test didn't identify the bug previously? Or phrased differently, are you choosing data that doesn't satisfy the constraint and therefore can be used to verify the constraint is being computed correctly?

Reply:
The random test already does that. 

But yes, I fixed the compute tag test to actually use the numerical spatial derivative, which won’t give zero for the constraint resolution is infinite

Reply:
I think it would be redundant, but if you’re not convinced, it would be easy enough to add a line computing the 3-index constraint from two random tensor was well...just let me know if you’d prefer that :)

Reply:
The random test  checks that the function works, not the tag, and it was the tag that was broken. However, if the test were compute Phi and d_i g_{ab} independently then the test should've failed, no? My concern is that the test didn't catch the fact that the constraint *could never be violated* since it basically ended up computing `Phi - Phi`. I think if phi were to be set to random values after things are added to the DataBox then it would've been clear that the compute tag was incorrect. Does that make sense?

Reply:
Yep! I’ll push a fixup that tests the tag with a random phi

Reply:
okay, just go ahead and squash it in :)

Reply:
The cleanest solution turned out to be making deriv_spacetime_metric random instead of phi, so that's what I went with.